src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -100,8 +100,14 @@ public static void initialize(ModelDao modelDao) {
     }
 
     private static float[] ObjectsToFloats(List<Object> objs) {
+        if (Objects.isNull(objs)) {
+            throw new IllegalArgumentException(String.format("[%s] requires 'vector' to be non-null", NAME));
+        }
         float[] vec = new float[objs.size()];
         for (int i = 0; i < objs.size(); i++) {
+            if (!(objs.get(i) instanceof Number)) {
+                throw new IllegalArgumentException(String.format("[%s] requires 'vector' to be an array of numbers", NAME));
```

Comment:
nit: may be,  field name: [%s] must contains array of numbers

Reply:
Updated